component/src/test/java/io/siddhi/extension/io/nats/sink/STANSinkTestCase.java

```diff
@@ -0,0 +1,420 @@
+        ResultContainer resultContainer = new ResultContainer(2, 3);
+        STANClient stanClient = new STANClient("test-cluster", "stan_test1", "nats://localhost:"
+                + port, resultContainer);
+        stanClient.connect();
+        SiddhiManager siddhiManager = new SiddhiManager();
+
+        String inStreamDefinition = "@App:name('Test-plan1')\n"
+                + "@sink(type='nats', @map(type='xml'), "
+                + "destination='nats-test1', "
+                + "client.id='test-plan1-siddhi',"
+                + "server.urls='" + "nats://localhost:" + port + "', "
+                + "streaming.cluster.id='test-cluster'"
+                + ")"
+                + "define stream inputStream (name string, age int, country string);";
+
+        stanClient.subsripeFromNow("nats-test1");
+        SiddhiAppRuntime executionPlanRuntime = siddhiManager.
+                createSiddhiAppRuntime(inStreamDefinition);
+        InputHandler inputStream = executionPlanRuntime.getInputHandler("inputStream");
+        executionPlanRuntime.start();
+
+        inputStream.send(new Object[]{"JAMES", 23, "USA"});
+        inputStream.send(new Object[]{"MIKE", 23, "Germany"});
+        Thread.sleep(500);
+        Assert.assertTrue(resultContainer.assertMessageContent("JAMES"));
+        Assert.assertTrue(resultContainer.assertMessageContent("MIKE"));
+        siddhiManager.shutdown();
+    }
+
+    /**
+     * if a property missing from the siddhi stan sink which defined as mandatory in the extension definition, then
+     * {@link SiddhiAppValidationException} will be thrown.
+     */
+    @Test(expectedExceptions = SiddhiAppValidationException.class,
+            dependsOnMethods = "natsSimplePublishTest")
+    public void testMissingNatsMandatoryProperty() {
+        SiddhiManager siddhiManager = new SiddhiManager();
+        String inStreamDefinition = "@App:name('Test-plan2')\n"
+                + "@sink(type='nats', @map(type='xml'), "
+                + "server.urls='" + "nats://localhost:" + port + "', "
+                + "client.id='test-plan2',"
+                + "cluster.id='test-cluster'"
+                + ")"
+                + "define stream inputStream (name string, age int, country string);";
+
+        siddhiManager.createSiddhiAppRuntime(inStreamDefinition);
+        siddhiManager.shutdown();
+    }
+
+    /**
+     * If invalid NATS url provided then {@link SiddhiAppValidationException} will be thrown.
+     */
+    @Test(expectedExceptions = SiddhiAppValidationException.class,
+            dependsOnMethods = "testMissingNatsMandatoryProperty")
+    public void testInvalidNatsUrl() {
+        SiddhiManager siddhiManager = new SiddhiManager();
+        String inStreamDefinition = "@App:name('Test-plan3')\n"
+                + "@sink(type='nats', @map(type='xml'), "
+                + "destination='nats-test3', "
+                + "bootstrap.servers='natss://localhost:4222',"
```

Comment:
We deprecated `bootstrap.servers`, right? So we need to replace it with the new convention in these tests?

Reply:
Fixed in https://github.com/siddhi-io/siddhi-io-nats/pull/37/commits/8ed0ca4c7792a9e1fe7c651b58d07f4548c6a02f